buildSrc/build.gradle.kts

```diff
@@ -50,12 +51,21 @@ subprojects {
             groovyOptions.encoding = "utf-8"
             configureCompileTask(this, options)
         }
+
+        val compileGroovy: GroovyCompile by tasks
+
+        configurations {
+            "apiElements" {
+                outgoing.variants["classes"].artifact(mapOf(
+                    "file" to compileGroovy.destinationDir,
+                    "type" to ArtifactTypeDefinition.JVM_CLASS_DIRECTORY,
+                    "builtBy" to compileGroovy
+                ))
+            }
```

Comment:
Is there any plan to make this automatic in the future when the `java-library` plugin is used in conjunction with the `groovy`/`scala`/`kotlin` plugin?

Reply:
It already seems work with the Kotlin plugin.
We should add that code for Groovy and Scala plugins I guess.
@melix Do we have plans to do so?